### DISCUSSION TITLE
The keyword delete is not working in Remix

### DISCUSSION THREAD

**Discussion by hsjo12** [remix-run/remix#2816] [Q&A] [answered]
Hello, 
I have came across the issue that the keyword delete is not working for mapping and array in remix, so I have tried to find why it is. 

In fact, if you put 0 into the mapping or array that has already values, it will get an error. 

For example, 
// array has 3 values
uint[] public array = [97,98,99];
// if I put 0 into the first index of the array, it will return an error
array[1] = 0; // error. 

// mapping 
mapping(string => uint) public map;   
//Not Error since it does not have any value yet other than default value 0
map["a"] = 0 

// Once the key set up the specific value 
map["a"] = 3;
//it wiill get an error from map["a"] = 0 
map["a"] = 0 

please try this code in remix, ( this code is totally working in truffle)

```
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.7.0 <0.9.0;

contract SimpleEx {

    uint[] public array = [97,98,99];
     
    function updateArrayZero(uint _index) public {
        array[_index] = 0;
    }

    function deleteArray(uint _index) public {
       delete array[_index];
    }

    mapping(string => uint) public map;   
    function Key_A_get_0() public {
        map["a"] = 0;
    }
    function Key_A_get_3() public {
        map["a"] = 3;
    }
    function delete_Key_A() public {
        delete map["a"];
    }
}
```

**Comment by machour** [comment, marked as answer]
You're on the wrong Remix repository, try https://github.com/ethereum/remix/

**Reply by hsjo12** [reply]
Thank you a lot :) , I will post it there
